src/v/cluster/log_eviction_stm.cc

```diff
@@ -291,6 +291,12 @@ ss::future<log_eviction_stm::offset_result> log_eviction_stm::replicate_command(
         }
     } catch (const ss::timed_out_error&) {
         result = errc::timeout;
+    } catch (...) {
+        vlog(
+          _log.warn,
+          "Replicating prefix_truncate failed with exception: {}",
+          std::current_exception());
+        result = errc::replication_error;
```

Comment:
just a quick question, is this going to be propagated to Kafka handler ?

Reply:
Yeah it eventually will be translated into a `kafka::error_code` here https://github.com/redpanda-data/redpanda/blob/dev/src/v/kafka/server/replicated_partition.cc#L399 then that is propagated back to the client in the delete_records handler.

Reply:
From the linked snippet, wouldn't this return a USE? Curious why that instead of explicitly handling the `is_shutdown_exception` case with `raft::errc::shutting_down` which results in a (I think) retriable timeout error?

Reply:
`consensus::replicate` will return `errc::shutting_down` as an error code not as an exception so that path is already covered.